pkg/lib/aws/ec2.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2019 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package aws
+
+import (
+	"math"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	s "github.com/cortexlabs/cortex/pkg/lib/strings"
+)
+
+// Returns the minium of all AZs in the region, 0 if unable to retreive spot price for any reason
+func SpotInstancePrice(accessKeyID string, secretAccessKey string, region string, instanceType string) float64 {
```

Comment:
I think I would prefer returning an error and allowing the user to handle it rather than having 0 as the error state